### PR TITLE
Added a datetime knob

### PR DESCRIPTION
**Pull request by madushan1000**
Datetime knob to test react components taking Date objects or timestamps as props. 
To use, import `date` from `storybook-addon-knobs` and use as `date('<name>', <default value>)`

### CODE REVIEW

**Comment by arunoda** [comment]
Do `npm test` and fix common issues.

File: src/components/types/Date.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  knob: React.PropTypes.object,
+  onChange: React.PropTypes.func,
+};
+
+DateType.serialize = function (value) {
+  return value;
+};
+
+DateType.deserialize = function (value) {
+  return value;
```
Comment (arunoda) [inline src/components/types/Date.js]: Don't we need to parseFloat here.

File: src/index.js [hunk 2]
```diff
@@ -27,6 +27,11 @@ export function select(name, options, value) {
   return manager.knob(name, { type: 'select', options, value });
 }
 
+export function date(name, value) {
+  const r =  manager.knob(name, { type: "date", value: value.toISOString() });
```
Comment (arunoda) [inline src/index.js]: I like to use a timestamp here. It's more generic in JS and smaller.
I think you need to change several places.

Comment (madushan1000) [inline src/index.js]: done

File: .scripts/mocha_runner.js [hunk 3]
```diff
@@ -11,6 +11,8 @@ require('babel-polyfill');
 var jsdom = require('jsdom').jsdom;
 
 var exposedProperties = ['window', 'navigator', 'document'];
+var fileExts = ['jpg', 'png', 'gif', 'eot', 'svg', 'ttf', 'woff', 'woff2'];
```
Comment (arunoda) [inline .scripts/mocha_runner.js]: I think this is because we included css.
So, just add a single entry for just handling `.css`

Comment (madushan1000) [inline .scripts/mocha_runner.js]: done

Comment (arunoda) [inline .scripts/mocha_runner.js]: Great. Thanks.

File: src/components/types/Date.css [hunk 4]
```diff
@@ -0,0 +1,217 @@
+.rdt {
```
Comment (arunoda) [inline src/components/types/Date.css]: It's pretty hard to load CSS specific to our manager's code at Storybook.
So, we need to think something else.

Here's something we can do.
See this library: https://github.com/substack/insert-css

We need to write this file as a `.js` file which returns a big string of CSS.
Then we can simply use above library to apply CSS.

File: .scripts/mocha_runner.js [hunk 5]
```diff
@@ -30,4 +30,8 @@ process.on('unhandledRejection', function (error) {
   console.error(error && error.stack || error);
 });
 
+require.extensions[`.css`] = function () {
```
Comment (arunoda) [inline .scripts/mocha_runner.js]: Once we added new CSS loading system. We don't need this.

File: package.json [hunk 6]
```diff
@@ -53,6 +53,8 @@
   "dependencies": {
     "babel-runtime": "^6.5.0",
     "deep-equal": "^1.0.1",
+    "moment": "^2.15.0",
```
Comment (arunoda) [inline package.json]: This should be a devDependancy as we only use this on the example.

File: src/components/types/Date.js [hunk 7]
```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import Datetime from 'react-datetime';
+import './Date.css';
```
Comment (arunoda) [inline src/components/types/Date.js]: Import the JS string of CSS here and use `insert-css` NPM module to load CSS.
If I doesn't work, try to do that in [`DOMContentLoaded`](https://developer.mozilla.org/en/docs/Web/Events/DOMContentLoaded) event.

File: package.json (hunk 6 above)
Comment (madushan1000) [inline package.json]: moment is a peer dependency of the react component I'm using. How should I handle that?

Comment (arunoda) [inline package.json]: Okay. If so, this is good.

File: src/components/types/Date.js (hunk 7 above)
Comment (madushan1000) [inline src/components/types/Date.js]: It's working properly now

**Pull request merged**